### PR TITLE
perf: optimize dockerfile

### PR DESCRIPTION
**Pull request by thebedigupta**
**Description**

Updated Test Results [test-results.md](https://github.com/user-attachments/files/17511811/test-results.md)

**Problems that I worked in this PR**

1. [Remove cache after installing packages ](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZB4sekEgviDxNeZYbKm)
2. [Merge this run instruction with the consecutive ones](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZFCmY8SYFR-KciwR6Ur)
3. [Surround this variable with double quotes; otherwise, it can lead to unexpected behavior](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZB4sekEgviDxNeZYbKn)
4. [Include the ARG instruction in the build stage where it is used](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZB4sekEgviDxNeZYbKo)

**I solved this issue after reading sonarcloud suggestion and how it work**

### CODE REVIEW

File: apps/generator/Dockerfile [hunk 1]
```diff
@@ -8,14 +6,17 @@ WORKDIR /app
 ENV PUPPETEER_EXECUTABLE_PATH /usr/bin/chromium-browser
 ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD true
 
+# Include the ARG instruction in this build stage
```
Comment (derberg) [inline apps/generator/Dockerfile:9]: please include one line comment explaining why it should be included in build stage

File: apps/generator/Dockerfile [hunk 2]
```diff
@@ -8,14 +6,17 @@ WORKDIR /app
...
 # Since 0.14.0 release of html-template chromium is needed for pdf generation.
 # More custom packages for specific template should not be added to this dockerfile. Instead, we should come up with some extensibility solution.
-RUN apk --update add git chromium && \
-    rm -rf /var/lib/apt/lists/* && \
-    rm /var/cache/apk/*
+# The --no-cache prevents from caching of the package index which saves space by not storing the APK cache files.
+# Install dependencies and AsyncAPI Generator in a single RUN command
+RUN apk --no-cache add git chromium && \
+    rm -rf /var/cache/apk/* && \
+    npm install -g --ignore-scripts "@asyncapi/generator@${ASYNCAPI_GENERATOR_VERSION}"
```
Comment (derberg) [inline apps/generator/Dockerfile:19]: sonar was just suggesting to surround only the variable name with double quotes

**Review by derberg — COMMENTED** [review]
once you react to all comments, please also run build of the docker image locally + use the docker to verify all works as expected

**Comment by thebedigupta** [comment]
> once you react to all comments, please also run build of the docker image locally + use the docker to verify all works as expected

I am updating test results file Please review it and then just tell me if I did something wrong

**Comment by derberg** [comment]
I see in test result that you did run `docker build` but haven't seen `docker run`. 

> use the docker to verify all works as expected

I mean't - please try to use the image you build locally, to check if generation still works

going with [this](https://github.com/asyncapi/generator/blob/master/Development.md?plain=1#L42) should do the trick

**Comment by thebedigupta** [comment]
> I see in test result that you did run `docker build` but haven't seen `docker run`.
> 
> > use the docker to verify all works as expected
> 
> I mean't - please try to use the image you build locally, to check if generation still works
> 
> going with [this](https://github.com/asyncapi/generator/blob/master/Development.md?plain=1#L42) should do the trick

You mean to say run that image which I build

**Comment by thebedigupta** [comment]
> I see in test result that you did run `docker build` but haven't seen `docker run`.
> 
> > use the docker to verify all works as expected
> 
> I mean't - please try to use the image you build locally, to check if generation still works
> 
> going with [this](https://github.com/asyncapi/generator/blob/master/Development.md?plain=1#L42) should do the trick

@derberg I was trying to understand where things goes wrong Give me some time if you have any suggestions on how to run image  then please let me know
docker

**Comment by derberg** [comment]
but the command that is under the link I provided, isn't it working?

**Comment by thebedigupta** [comment]
> but the command that is under the link I provided, isn't it working?

Update the test results Please check again

**Comment by derberg** [comment]
@thebedigupta you mean in description? I don't see info in test result that `docker run` was called

**Comment by derberg** [comment]
no follow-up from contributor - closing PR, we do not want to keep such inactive PRs open

you can restart the work anytime you want, you or any other contributor - just feel free to open another PR. You see previous comments, you know what has to be improved

**Pull request closed without merging**